src/realm/group_shared.cpp

```diff
@@ -749,6 +749,30 @@ std::string SharedGroupOptions::sys_tmp_dir = "";
 // initializing process crashes and leaves the shared memory in an
 // undefined state.
 
+bool SharedGroup::delete_realm(const std::string path, cleanup_func cleanup) {
+    // auto coordination_dir = path + ".management";
+    auto lockfile_path(path + ".lock");
+
+    File lockfile;
+    lockfile.open(lockfile_path, File::access_ReadWrite, File::create_Auto, 0); // Throws
+    File::CloseGuard fcg(lockfile);
+
+    if (lockfile.try_lock_exclusive()) { // Throws
+        File::UnlockGuard ulg(lockfile);
+        std::remove(path.c_str()); // ignore result, file may not exist
+
+        // TODO: remove coordination directory.
+
+        // allow caller to do furter cleanup under lock
+        if (cleanup)
+            cleanup(path);
```

Comment:
The path should probably point to the directory instead of file as that would be more useful for implementors

Reply:
what's the point of the cleanup function if it's only run when the function returns true?

Reply:
Ok - that only makes sense if the call to the cleanup function goes before deleting the directory, so I'll change the ordering. My idea was to wipe the directory and delete it, so that the cleanup function was only necessary for files/resource outside of the directory.

Reply:
But the `path` parameter is to the realm file, right. and you are only cleaning the special folder `path + .management` I assume, because Core cleaning the whole directory which might contain non-related files, would be a big no no.

Reply:
Oh, you mean the directory holding the realm file? That would not hold enough information. What if people had multiple realms in the same directory? The cleanup function would need to act on something that included the full path.

Reply:
But it might be better to provide a way for the caller to pass information through into the callback. That'd give more flexibility.

Reply:
Good point. The `path` as it is now is probably fine, since all other helper files just add a prefix to that if I recall correctly. 

```
// Our currentl logic for deleting old style Realms
                new File(rootFolder, realmFileName + ".log_a"),
                new File(rootFolder, realmFileName + ".log_b"),
                new File(rootFolder, realmFileName + ".log"),
```

Reply:
the full path to the realm file is good! 👍

Reply:
Hmmm - I'm going to change it into taking a function object (lambda) instead.